Building on aarch64-pc-windows-msvc

**Is your feature request related to a problem? Please describe.**

I want to be able to build and run kanidm for `aarch64-pc-windows-msvc` using native binaries (not emulation).

At present, `cargo build` fails with:

```
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `C:\Users\michael\Documents\dev\kanidm\target\debug\build\ring-9d8a42b05c7b5a76\build-script-build` (exit code: 101)
  --- stdout
  OPT_LEVEL = Some("0")
  TARGET = Some("aarch64-pc-windows-msvc")
  HOST = Some("aarch64-pc-windows-msvc")
  cargo:rerun-if-env-changed=CC_aarch64-pc-windows-msvc
  CC_aarch64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CC_aarch64_pc_windows_msvc
  CC_aarch64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64-pc-windows-msvc
  CFLAGS_aarch64-pc-windows-msvc = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_pc_windows_msvc
  CFLAGS_aarch64_pc_windows_msvc = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("neon,pmuv3")
  DEBUG = Some("true")
  montgomery.c
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(191): error C2065: 'BN_ULLONG': undeclared identifier
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(191): error C2146: syntax error: missing ';' before identifier 'result'
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(191): warning C4555: result of expression not used
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(191): error C2065: 'result': undeclared identifier
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(191): error C2146: syntax error: missing ';' before identifier 'a'
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(191): warning C4552: '*': result of expression not used
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(192): error C2065: 'result': undeclared identifier
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(193): error C2065: 'result': undeclared identifier
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\internal.h(193): warning C4293: '>>': shift count negative or too big, undefined behavior
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(38): warning C4163: '_addcarry_u64': not available as an intrinsic function
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(38): warning C4163: '_subborrow_u64': not available as an intrinsic function
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(62): warning C4013: '_addcarry_u64' undefined; assuming extern returning int
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(62): warning C4242: '=': conversion from 'int' to 'Carry', possible loss of data
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(76): warning C4242: '=': conversion from 'int' to 'Carry', possible loss of data
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(92): warning C4013: '_subborrow_u64' undefined; assuming extern returning int
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(92): warning C4242: '=': conversion from 'int' to 'Carry', possible loss of data
  C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\crypto\fipsmodule\bn\../../limbs/limbs.inl(106): warning C4242: '=': conversion from 'int' to 'Carry', possible loss of data

  --- stderr
  running "C:\\Program Files\\Microsoft Visual Studio\\2022\\Preview\\VC\\Tools\\MSVC\\14.34.31933\\bin\\HostX86\\arm64\\cl.exe" "-nologo" "-MD" "-Z7" "-Brepro" "-I" "include" "-W4" "/GS" "/Gy" "/EHsc" "/GR-" "/Zc:wchar_t" "/Zc:forScope" "/Zc:inline" "/Zc:rvalueCast" "/sdl" "/Wall" "/wd4127" "/wd4464" "/wd4514" "/wd4710" "/wd4711" "/wd4820" "/wd5045" "/Od" "/RTCsu" "-DNDEBUG" "-c" "/FoC:\\Users\\michael\\Documents\\dev\\kanidm\\target\\debug\\build\\ring-9a630ec9f0801d55\\out\\montgomery.obj" "crypto/fipsmodule/bn/montgomery.c"
  thread 'main' panicked at 'execution failed', C:\Users\michael\.cargo\registry\src\github.com-1ecc6299db9ec823\ring-0.16.20\build.rs:656:9

> cargo tree -i ring
ring v0.16.20
├── rustls v0.20.7
│   ├── hyper-rustls v0.23.1
│   │   └── reqwest v0.11.12
│   │       ├── kanidm_client v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidm_client)
│   │       │   ├── kanidm-ipa-sync v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\iam_migrations\freeipa)
│   │       │   ├── kanidm_tools v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidm_tools)
│   │       │   ├── kanidm_unix_int v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidm_unix_int)
│   │       │   │   ├── nss_kanidm v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidm_unix_int\nss_kanidm)
│   │       │   │   └── pam_kanidm v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidm_unix_int\pam_kanidm)
│   │       │   ├── kanidmd_testkit v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidmd\testkit)
│   │       │   └── orca v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\orca)
│   │       ├── kanidm_unix_int v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidm_unix_int) (*)
│   │       ├── kanidmd_testkit v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidmd\testkit)
│   │       └── oauth2 v4.3.0
│   │           └── kanidmd_testkit v1.1.0-alpha.11-dev (C:\Users\michael\Documents\dev\kanidm\kanidmd\testkit)
│   ├── reqwest v0.11.12 (*)
│   └── tokio-rustls v0.23.4
│       ├── hyper-rustls v0.23.1 (*)
│       └── reqwest v0.11.12 (*)
├── sct v0.7.0
│   └── rustls v0.20.7 (*)
└── webpki v0.22.0
    ├── rustls v0.20.7 (*)
    ├── tokio-rustls v0.23.4 (*)
    └── webpki-roots v0.22.5
        └── reqwest v0.11.12 (*)
```

It appears this _specific_ failure is a problem with `ring`, which is an indirect dependency of `kanidm`; though there may be other issues I haven't found yet.

I'm making an issue mainly so that I can keep track of things until this builds, and the docs are up to date. :)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -97,7 +97,7 @@ libsqlite3-sys = "^0.25.0"
 lru = "^0.8.0"
 mathru = "^0.13.0"
 num_enum = "^0.5.7"
-oauth2_ext = { version = "^4.1.0", package = "oauth2" }
+oauth2_ext = { version = "^4.1.0", package = "oauth2", default-features = false }
 openssl-sys = "^0.9"
 openssl = "^0.10.41"
 paste = "^1.0.9"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -110,7 +110,7 @@ r2d2 = "^0.8.9"
 r2d2_sqlite = "^0.21.0"
 rand = "^0.8.5"
 regex = "1.7.0"
-reqwest = "0.11.13"
+reqwest = { version = "0.11.13", default-features = false, features=["cookies", "json", "gzip", "native-tls"] }
 rpassword = "^7.1.0"
 rusqlite = "^0.28.0"
 saffron = "^0.1.0"

--- a/kanidm_client/Cargo.toml
+++ b/kanidm_client/Cargo.toml
@@ -13,7 +13,7 @@ repository.workspace = true
 
 [dependencies]
 tracing.workspace = true
-reqwest = { workspace = true, features=["cookies", "json", "native-tls"] }
+reqwest.workspace = true
 kanidm_proto.workspace = true
 serde = { workspace = true, features = ["derive"] }
 serde_json.workspace = true

--- a/kanidm_client/Cargo.toml
+++ b/kanidm_client/Cargo.toml
@@ -13,7 +13,7 @@ repository.workspace = true
 
 [dependencies]
 tracing.workspace = true
-reqwest.workspace = true
+reqwest = { workspace = true, default-features = false }
 kanidm_proto.workspace = true
 serde = { workspace = true, features = ["derive"] }
 serde_json.workspace = true

--- a/kanidm_unix_int/Cargo.toml
+++ b/kanidm_unix_int/Cargo.toml
@@ -68,7 +68,7 @@ toml.workspace = true
 tokio = { workspace = true, features = ["rt", "macros", "sync", "time", "net", "io-util"] }
 tokio-util = { workspace = true, features = ["codec"] }
 tracing.workspace = true
-reqwest.workspace = true
+reqwest = { workspace = true, default-features = false }
 users.workspace = true
 walkdir.workspace = true
 

--- a/kanidmd/lib/src/filter.rs
+++ b/kanidmd/lib/src/filter.rs
@@ -881,6 +881,22 @@ impl FilterComp {
                 admin_error!("Unable to convert ldapsubstringfilter to sub filter");
                 return Err(OperationError::FilterGeneration);
             }
+            LdapFilter::GreaterOrEqual(_, _) => {
+                admin_error!("Unsupported filter operation - greater or equal");
+                return Err(OperationError::FilterGeneration);
+            }
+            LdapFilter::LessOrEqual(_, _) => {
+                admin_error!("Unsupported filter operation - less or equal");
+                return Err(OperationError::FilterGeneration);
+            }
+            LdapFilter::Approx(_, _) => {
+                admin_error!("Unsupported filter operation - approximate");
+                return Err(OperationError::FilterGeneration);
+            }
+            LdapFilter::Extensible(_) => {
+                admin_error!("Unsupported filter operation - extensible");
+                return Err(OperationError::FilterGeneration);
+            }
         })
     }
 }

--- a/kanidmd/testkit/Cargo.toml
+++ b/kanidmd/testkit/Cargo.toml
@@ -26,7 +26,7 @@ webauthn-authenticator-rs.workspace = true
 oauth2_ext = { workspace = true, default-features = false }
 
 url = { workspace = true, features = ["serde"] }
-reqwest = { workspace = true, features=["cookies", "json", "native-tls"] }
+reqwest.workspace = true
 sketching.workspace = true
 testkit-macros.workspace = true
 tracing = { workspace = true, features = ["attributes"] }

--- a/kanidmd/testkit/Cargo.toml
+++ b/kanidmd/testkit/Cargo.toml
@@ -26,7 +26,7 @@ webauthn-authenticator-rs.workspace = true
 oauth2_ext = { workspace = true, default-features = false }
 
 url = { workspace = true, features = ["serde"] }
-reqwest.workspace = true
+reqwest = { workspace = true, default-features = false }
 sketching.workspace = true
 testkit-macros.workspace = true
 tracing = { workspace = true, features = ["attributes"] }

